pipeline_dp/aggregate_params.py

```diff
@@ -587,6 +586,35 @@ class PrivacyIdCountParams:
     contribution_bounds_already_enforced: bool = False
 
 
+@dataclass
+class AddDPNoiseParams:
+    """Specifies parameters for function DPEngine.add_dp_noise()
+
+   Attributes:
+       noise_kind: The type of noise to use for the DP calculations.
+       l0_sensitivity: the maximum number of partition for which 1 privacy unit
+         can contribute.
+       linf_sensitivity: the maximum difference of values in one partition which
+         can achieved by adding or removing one privacy unit from the dataset.
+       budget_weight: Relative weight of the privacy budget allocated to this
+         aggregation.
+   """
+    noise_kind: NoiseKind
+    l0_sensitivity: int
+    linf_sensitivity: float
+    budget_weight: float = 1
+
+    def __post_init__(self):
```

Comment:
perhaps for a different change: should we add similar checks to other Params classes?